File: typescript/questions/maxProfit2.ts
function maxProfit(prices: number[]): number {
  let mp = 0;
  for (let p = 1; p < prices.length; ++p) {
    const cp = prices[p] - prices[p - 1];
    mp += cp > 0 ? cp : 0;
  }
  return mp;
}

File: python/questions/maxProfit2.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        mp = 0
        for i in range(1, len(prices)):
            cp = prices[i] - prices[i - 1]
            mp += cp if cp > 0 else 0
        return mp
